src/libraries/System.Formats.Tar/src/System/Formats/Tar/TarWriter.cs

```diff
@@ -365,5 +369,16 @@ private async ValueTask WriteFinalRecordsAsync()
 
             return (fullPath, actualEntryName);
         }
+
+        private static void ValidateEntryLinkName(TarEntryType entryType, string? linkName)
```

Comment:
This is used in one place.  Why not just add it to the call site rather than separating out the helper method?

Reply:
Hmm, ok.